### ISSUE TITLE
where is node-v11-linux-x64.tar.gz ?

### ISSUE THREAD

**Issue by jsperrazza** [libxmljs/libxmljs#463]
Hello,

We were using https://github.com/libxmljs/libxmljs/releases/download/v0.18.4/node-v11-linux-x64.tar.gz and now we are getting a 404 error. Do you know where node-v11-linux-x64.tar.gz was moved to or where the latest version is located? We are unable to create a new Hubot RPM because of the dependency. We are receiving the below error message. 

make: Leaving directory `/builddir/build/BUILD/vbot-2.11.5/node_modules/nodepie/node_modules/xml2json/node_modules/node-expat/build'
> libxmljs@0.18.4 install /builddir/build/BUILD/vbot-2.11.5/node_modules/wolfram/node_modules/libxmljs
> node-pre-gyp install --fallback-to-build --loglevel http
node-pre-gyp http GET https://github.com/libxmljs/libxmljs/releases/download/v0.18.4/node-v11-linux-x64.tar.gz
node-pre-gyp http 404 https://github.com/libxmljs/libxmljs/releases/download/v0.18.4/node-v11-linux-x64.tar.gz
node-pre-gyp ERR! Tried to download(404): https://github.com/libxmljs/libxmljs/releases/download/v0.18.4/node-v11-linux-x64.tar.gz 
node-pre-gyp ERR! Pre-built binaries not found for libxmljs@0.18.4 and node@0.10.32 (node-v11 ABI) (falling back to source compile with node-gyp) 
node-pre-gyp http 404 status code downloading tarball https://github.com/libxmljs/libxmljs/releases/download/v0.18.4/node-v11-linux-x64.tar.gz 
module.js:340
    throw err;
 
 
 
Error: Cannot find module 'graceful-fs'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/app/nodejs/lib/node_modules/node-gyp/lib/node-gyp.js:12:10)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/app/nodejs/bin/node /app/nodejs/lib/node_modules/node-gyp/bin/node-gyp.js clean' ( 
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/builddir/build/BUILD/vbot-2.11.5/node_modules/wolfram/node_modules/libxmljs/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:98:17)
node-pre-gyp ERR! stack     at maybeClose (child_process.js:756:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:823:5)
node-pre-gyp ERR! System Linux 2.6.32-358.2.1.el6.x86_64
node-pre-gyp ERR! command "node" "/builddir/build/BUILD/vbot-2.11.5/node_modules/wolfram/node_modules/libxmljs/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build" "--loglevel" "http"
node-pre-gyp ERR! cwd /builddir/build/BUILD/vbot-2.11.5/node_modules/wolfram/node_modules/libxmljs
node-pre-gyp ERR! node -v v0.10.32
node-pre-gyp ERR! node-pre-gyp -v v0.6.34
node-pre-gyp ERR! not ok

**Comment by rchipka** [comment]
The file `node-v11-linux-x64.tar.gz` does not exist.

This library doesn't have pre-built binaries for linux yet, and instead requires building.

The option `--fallback-to-build` should force libxmljs to be built if it can't find the appropriate binary.

I'm looking into this issue.

**Comment by jsperrazza** [comment]
Good morning,

Thanks for the information. I’ll try the fall back to build option.

Jamie

[cid:F5BC8804-8F24-4C9F-ACC3-78DD76453F76][Verisign]

Jamie Sperrazza
Software Engineer III
jsperrazza@Verisign.com

Desk: 302-395-4959
Cell: 703-835-6240
21 Boulden Circle
New Castle  DE  19720
UNITED STATES
VerisignInc.com<http://www.verisigninc.com/>

“Success in life comes when you
Simply refuse to give up, with
Goals so strong that Obstacles,
Failures, and Loss only act as
Motivation”

**Comment by pberson** [comment]
I'm also getting this issue.
node-pre-gyp http GET https://github.com/libxmljs/libxmljs/releases/download/v0.18.4/node-v51-linux-x64.tar.gz
node-pre-gyp http 404 https://github.com/libxmljs/libxmljs/releases/download/v0.18.4/node-v51-linux-x64.tar.gz
node-pre-gyp ERR! Tried to download(404): https://github.com/libxmljs/libxmljs/releases/download/v0.18.4/node-v51-linux-x64.tar.gz 
node-pre-gyp ERR! Pre-built binaries not found for libxmljs@0.18.4 and node@7.8.0 (node-v51 ABI) (falling back to source compile with node-gyp) 
node-pre-gyp http 404 status code downloading tarball https://github.com/libxmljs/libxmljs/releases/download/v0.18.4/node-v51-linux-x64.tar.gz

**Comment by andreg** [comment]
Same here, not sure how to proceed.

**Comment by rchipka** [comment]
One solution would be to run `node-gyp rebuild` inside the libxmljs directory.

**Comment by GarrettKucinski** [comment]
Getting an error on falling-back-to-build

```
> libxmljs@0.18.4 install /Users/Garrett/Desktop/LAB/js/javascript_techdegree/node-content-scraper/node_modules/libxmljs
> node-pre-gyp install --fallback-to-build --loglevel http

node-pre-gyp http GET https://github.com/libxmljs/libxmljs/releases/download/v0.18.4/node-v57-darwin-x64.tar.gz
node-pre-gyp http 404 https://github.com/libxmljs/libxmljs/releases/download/v0.18.4/node-v57-darwin-x64.tar.gz
node-pre-gyp ERR! Tried to download(404): https://github.com/libxmljs/libxmljs/releases/download/v0.18.4/node-v57-darwin-x64.tar.gz
node-pre-gyp ERR! Pre-built binaries not found for libxmljs@0.18.4 and node@8.0.0 (node-v57 ABI) (falling back to source compile with node-gyp)
node-pre-gyp http 404 status code downloading tarball https://github.com/libxmljs/libxmljs/releases/download/v0.18.4/node-v57-darwin-x64.tar.gz
```

Any thoughts on this? Not sure what do do about it?
```
In file included from ../vendor/libxml/xpath.c:476:
../vendor/libxml/trionan.c:194:1: warning: unused function 'trio_is_special_quantity' [-Wunused-function]
trio_is_special_quantity
^
../vendor/libxml/trionan.c:218:1: warning: unused function 'trio_is_negative' [-Wunused-function]
trio_is_negative
^
2 warnings generated.
  CC(target) Release/obj.target/libxml/vendor/libxml/xpointer.o
  LIBTOOL-STATIC Release/xml.a
Traceback (most recent call last):
  File "./gyp-mac-tool", line 611, in <module>
    sys.exit(main(sys.argv[1:]))
  File "./gyp-mac-tool", line 28, in main
    exit_code = executor.Dispatch(args)
  File "./gyp-mac-tool", line 43, in Dispatch
    return getattr(self, method)(*args[1:])
  File "./gyp-mac-tool", line 246, in ExecFilterLibtool
    if not libtool_re.match(line) and not libtool_re5.match(line):
TypeError: cannot use a string pattern on a bytes-like object
make: *** [Release/xml.a] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/Garrett/.nvm/versions/node/v8.0.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:285:23)
gyp ERR! stack     at emitTwo (events.js:125:13)
gyp ERR! stack     at ChildProcess.emit (events.js:213:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:197:12)
gyp ERR! System Darwin 16.6.0
gyp ERR! command "/Users/Garrett/.nvm/versions/node/v8.0.0/bin/node" "/Users/Garrett/.nvm/versions/node/v8.0.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--loglevel=http" "--module=/Users/Garrett/Desktop/LAB/js/javascript_techdegree/node-content-scraper/node_modules/libxmljs/build/Release/xmljs.node" "--module_name=xmljs" "--module_path=/Users/Garrett/Desktop/LAB/js/javascript_techdegree/node-content-scraper/node_modules/libxmljs/build/Release"
gyp ERR! cwd /Users/Garrett/Desktop/LAB/js/javascript_techdegree/node-content-scraper/node_modules/libxmljs
gyp ERR! node -v v8.0.0
gyp ERR! node-gyp -v v3.6.1
gyp ERR! not ok
node-pre-gyp ERR! build error
node-pre-gyp ERR! stack Error: Failed to execute '/Users/Garrett/.nvm/versions/node/v8.0.0/bin/node /Users/Garrett/.nvm/versions/node/v8.0.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --loglevel=http --module=/Users/Garrett/Desktop/LAB/js/javascript_techdegree/node-content-scraper/node_modules/libxmljs/build/Release/xmljs.node --module_name=xmljs --module_path=/Users/Garrett/Desktop/LAB/js/javascript_techdegree/node-content-scraper/node_modules/libxmljs/build/Release' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/Users/Garrett/Desktop/LAB/js/javascript_techdegree/node-content-scraper/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at emitTwo (events.js:125:13)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:213:7)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:887:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:208:5)
node-pre-gyp ERR! System Darwin 16.6.0
node-pre-gyp ERR! command "/Users/Garrett/.nvm/versions/node/v8.0.0/bin/node" "/Users/Garrett/Desktop/LAB/js/javascript_techdegree/node-content-scraper/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build" "--loglevel" "http"
node-pre-gyp ERR! cwd /Users/Garrett/Desktop/LAB/js/javascript_techdegree/node-content-scraper/node_modules/libxmljs
node-pre-gyp ERR! node -v v8.0.0
node-pre-gyp ERR! node-pre-gyp -v v0.6.36
node-pre-gyp ERR! not ok
Failed to execute '/Users/Garrett/.nvm/versions/node/v8.0.0/bin/node /Users/Garrett/.nvm/versions/node/v8.0.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --loglevel=http --module=/Users/Garrett/Desktop/LAB/js/javascript_techdegree/node-content-scraper/node_modules/libxmljs/build/Release/xmljs.node --module_name=xmljs --module_path=/Users/Garrett/Desktop/LAB/js/javascript_techdegree/node-content-scraper/node_modules/libxmljs/build/Release' (1)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! libxmljs@0.18.4 install: `node-pre-gyp install --fallback-to-build --loglevel http`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the libxmljs@0.18.4 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
```

**Comment by kirkins** [comment]
I have a project that was relying on this library and now I'm getting this same issue.

How would I use `--fallback-to-build`, I tried `npm install --fallback-to-build`, I'm not sure when to use this?

What would be needed to add pre-built binaries?

**Comment by pietermees** [comment]
I'm also interested in pre-built binaries for Linux. Maybe we could outline the work needed to accomplish this?

**Comment by rchipka** [comment]
Should be fixed by #470. Waiting for the builds to finish then will `npm publish`.
